service/history/execution/mutable_state_builder_methods_decision_test.go

```diff
@@ -0,0 +1,908 @@
+	builder := setupMutableStateBuilder(t)
+
+	decision := &DecisionInfo{}
+	identity := "identity"
+	rets := &types.HistoryEvent{}
+
+	builder.decisionTaskManager.(*MockmutableStateDecisionTaskManager).EXPECT().CreateTransientDecisionEvents(decision, identity).Return(rets, rets).Times(1)
+
+	historyEvent0, historyEvent1 := builder.CreateTransientDecisionEvents(decision, identity)
+
+	assert.Equal(t, rets, historyEvent0)
+	assert.Equal(t, rets, historyEvent1)
+}
+
+func TestCheckResettableMutableStateBuilder(t *testing.T) {
+	tests := []struct {
+		name                         string
+		pendingChildExecutionInfoIDs map[int64]*persistence.ChildExecutionInfo
+		pendingRequestCancelInfoIDs  map[int64]*persistence.RequestCancelInfo
+		pendingSignalInfoIDs         map[int64]*persistence.SignalInfo
+		wantErr                      bool
+		errMessage                   string
+	}{
+		{
+			name: "success",
+		},
+		{
+			name: "error due to pending child execution",
+			pendingChildExecutionInfoIDs: map[int64]*persistence.ChildExecutionInfo{
+				1: {},
+			},
+			wantErr:    true,
+			errMessage: "it is not allowed resetting to a point that workflow has pending child types.",
+		},
+		{
+			name: "error due to pending request cancel external",
+			pendingRequestCancelInfoIDs: map[int64]*persistence.RequestCancelInfo{
+				1: {},
+			},
+			wantErr:    true,
+			errMessage: "it is not allowed resetting to a point that workflow has pending request cancel.",
+		},
+		{
+			name: "error due to pending signal external",
+			pendingSignalInfoIDs: map[int64]*persistence.SignalInfo{
+				1: {},
+			},
+			wantErr:    true,
+			errMessage: "it is not allowed resetting to a point that workflow has pending signals to send.",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			builder := &mutableStateBuilder{
+				pendingChildExecutionInfoIDs: tc.pendingChildExecutionInfoIDs,
+				pendingRequestCancelInfoIDs:  tc.pendingRequestCancelInfoIDs,
+				pendingSignalInfoIDs:         tc.pendingSignalInfoIDs,
+			}
+
+			err := builder.CheckResettable()
+
+			if tc.wantErr {
+				assert.Error(t, err)
+				assert.Equal(t, tc.errMessage, err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+
+}
+
+func TestAddDecisionTaskCompletedEventMutableStateBuilder(t *testing.T) {
+	tests := []struct {
+		name          string
+		executionInfo *persistence.WorkflowExecutionInfo
+		wantErr       bool
+	}{
+		{
+			name: "success",
+			executionInfo: &persistence.WorkflowExecutionInfo{
+				State: persistence.WorkflowStateCreated,
+			},
+		},
+		{
+			name: "error due to state",
+			executionInfo: &persistence.WorkflowExecutionInfo{
+				State: persistence.WorkflowStateCompleted,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			builder := setupMutableStateBuilder(t)
+			builder.executionInfo = tc.executionInfo
+			builder.logger = loggerimpl.NewNopLogger()
```

Comment:
another indicator of not needing a `setupMutableStateBuilder` as a separate helper. you can just create in each test